src/main/scala/mesosphere/mesos/ResourceMatcher.scala

```diff
@@ -38,17 +38,15 @@ object ResourceMatcher {
     * accident.
     *
     * @param acceptedRoles contains all Mesos resource roles that are accepted
-    * @param reserved if reserved is true, only resources with a ReservationInfo
-    *                 are considered. If reserved is false, only resources without
-    *                 a ReservationInfo are considered.
-    * @param requiredLabels only resources with the given keys/values are matched.
+    * @param reservation if given, only resources with a ReservationInfo are
+    *                    considered and will only match if their labels match
+    *                    the specified labels.
     */
-  case class ResourceSelector(
-      acceptedRoles: Set[String], reserved: Boolean, requiredLabels: ResourceLabels = ResourceLabels.empty) {
+  case class ResourceSelector(acceptedRoles: Set[String], reservation: Option[ReservationSelector] = None) {
```

Comment:
Instead of having an optional ReservationSelector, we could implement 2 selectors:
1) that selects based on the existing labels
2) that selects based on non existing labels
But it is probably minor.